### DISCUSSION TITLE
https://github.com/pymupdf/PyMuPDF/labels/bug

### DISCUSSION THREAD

**Discussion by lchade** [pymupdf/PyMuPDF#1549] [Q&A]
## Describe the bug (mandatory)
The frames created by addRedactAnnot() will disappear after right-click.

## To Reproduce (mandatory)
We use this package to capture and annotate entities through the text. We use this statement to annotate the text that we want to apply the redact in the future:
> page.addRedactAnnot(area, text=row["class"], cross_out=False, fill=(0.75, 0.8, 0.95), text_color=(1,1,1))
("area" is a fitz.Rect object.)

Initially, the annotation result looks fine (see test_annotation.pdf attached) in Adobe Reader and Adobe Acrobat: 
![image](https://user-images.githubusercontent.com/44476697/150025649-06fdcad1-17d6-4e90-9a52-95594aa53740.png)
However, after we right-click the annotation frame (to view its property or apply redaction, etc.), the frame disappeared.
- right lick to see properties:
![Screenshot (4)](https://user-images.githubusercontent.com/44476697/150025972-27cbfdbd-ebd7-408a-94e6-315b66eef72c.png)
- the frame disappeared:
![image](https://user-images.githubusercontent.com/44476697/150026059-cf4e1470-6230-46ab-aef9-fd1295212529.png)
 
But when we hang the mouse over the text where the annotation frame is disappeared, it will show the annotate label and the fill color:
![Screenshot (2)](https://user-images.githubusercontent.com/44476697/150026333-1f309a92-a8f6-4a19-ad84-4f10bf708d04.png)

That means the annotation is actually still there, but the frame just somehow disappeared.
In the last line of this test pdf, I annotate the phrase manually through Adobe Acrobat:
![image](https://user-images.githubusercontent.com/44476697/150024323-4cebe2d9-115f-4f14-bf72-997daa8c7b0e.png)
In this way, its frame doesn't disappear. We want every frame processed by our code to stay still like this one.

Can you please advise how to prevent the frames processed by the program from disappearing? Thanks.

## Your configuration (mandatory)
 - Windows 10 64bit
 - Python 3.9
 - PyMuPDF 1.19.4
 
## Attachments
[test_annotation.pdf](https://github.com/pymupdf/PyMuPDF/files/7892532/test_annotation.pdf)

**Comment by JorjMcKie** [comment]
This has nothing to do with (Py-) MuPDF. This the behaviour of Adobe viewers, which I cannot influence.
All viewers have their own specialized way to deal with annotation (and form field) details. There are differences between Adobe and e.g. PDF XChange also.
In addition free software version also differ as to whether support all annotation types at all

**Comment by JorjMcKie** [comment]
I can confirm that when you right-click and select `properties` with Adobe, then the annotation is already changed be given a format as how Adobe thinks redactions should be coded.
And they can do that because there is no standard way how this should be. done.

**Reply by JorjMcKie** [reply]
When you do this and click ok, then Adobe has already changed the annot. Verify this looking at the resp. symbol which is no longer grayed out.
And that marks the end of PyMuPDF's responsibility.

**Reply by jeff-ywang** [reply]
Even when you click "Cancel" Adobe still changes the annotation anyway, which is a very strange behavior. However a redact annotation created directly with Adobe Pro doesn't seem to have the same issue. Is it possible that there is a gap between your implementation and Adobe's specification around redact annotation?

**Reply by JorjMcKie** [reply]
It is no gap - it is a difference.
There is no prescribed way how to implement annotations - no matter what type.
Literally every software does it differently. Therefore I recommended in my documentation to stick with one product exclusively for annot handling.
You simply cannot expect that things continue to work seemlessly if you create an annot (or field) with one product and then somehow maintain it with another one.
BTW when looking at what Adobe does after right clicking does not look good or consistent either - for example when you look at how the overlay text is rendered.
So my recommendations is do everything with Adobe ... or nothing.

**Reply by jeff-ywang** [reply]
Thanks for your reply. Unfortunately our use case requires leveraging AI to detect and add redact annotations on the fly (we've been using your solution for all pdf manipulations, fantastic product, thank you for the good work), and business will be using Adobe Pro to review/confirm the annotations and apply the final redaction.

**Reply by JorjMcKie** [reply]
@jeff-ywang -  I perfectly understand - and thanks for your appreciation of PyMuPDF.
